Assets/Scripts/GridGen.cs

```diff
@@ -174,6 +185,32 @@ public class GridGen : MonoBehaviour
         PutSprites();
     }
 
+    public void AddState() {
+        List<List<char>> tempList = new List<List<char>>();
+        for (int x = 0; x < rowsLength; x++) {
+            tempList.Add(new List<char>());
+            for (int y = 0; y < columnsLength; y++) {
+                tempList[x].Add(tiles[x][y]);
+            }
+        }
+
+        print("AddState " + states.Count);
+        states.Push(tempList);
+    }
+
+    public void RevertState() {
+        //Might need an additional pop if same
```

Comment:
Personally I prefer having tasks listed in a ticketing system or at least separately than the source code so that I can have a better centralized view of all tasks left to do.